### PR TITLE
Fix #2119: Handle I/O errors when creating files (export, publish etc)

### LINKED ISSUE #2119: Handle I/O errors when creating files (export, publish etc)
Need to review the code that creates files, and handle `IOException` as needed.

### PR DESCRIPTION
**Pull request by int19h**
Handle or wrap/rethrow IOException as appropriate.

Add special handling for T=byte[] in EvaluateAsync<T>for blobs.

### CODE REVIEW

File: src/Host/Client/Test/IdeGraphicsDeviceTest.cs [hunk 1]
```diff
@@ -577,8 +577,8 @@ public class IdeGraphicsDeviceTest {
 
         private async Task ExportToImageAsync(IRSession session, string format, string filePath, int widthInPixels,int heightInPixels, int resolution) {
             string script = String.Format("rtvs:::export_to_image({0}, {1}, {2}, {3})",format, widthInPixels, heightInPixels, resolution);
-            var result = await session.EvaluateAsync(script, REvaluationKind.RawResult);
-            result.SaveRawDataToFile(filePath);
+            var data = await session.EvaluateAsync<byte[]>(script, REvaluationKind.Normal);
+            File.WriteAllBytes(filePath, data);
```
Comment (MikhailArkhipov) [inline src/Host/Client/Test/IdeGraphicsDeviceTest.cs]: You may want IFileSystem for testability

Comment (MikhailArkhipov) [inline src/Host/Client/Test/IdeGraphicsDeviceTest.cs]: NVM, it is test code

File: src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs [hunk 2]
```diff
@@ -68,19 +68,19 @@ internal static class CsvAppFileIO {
...
             await TaskUtilities.SwitchToBackgroundThread();
 
             var sep = CultureInfo.CurrentCulture.TextInfo.ListSeparator;
             var dec = CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator;
 
             using (var e = await session.BeginEvaluationAsync()) {
-                var csvdata = await e.EvaluateAsync($"rtvs:::export_to_csv({result.Expression}, sep={sep.ToRStringLiteral()}, dec={dec.ToRStringLiteral()})", REvaluationKind.RawResult);
-                csvdata.SaveRawDataToFile(file);
+                var csvData = await e.EvaluateAsync<byte[]>($"rtvs:::export_to_csv({result.Expression}, sep={sep.ToRStringLiteral()}, dec={dec.ToRStringLiteral()})", REvaluationKind.Normal);
+                File.WriteAllBytes(fileName, csvData);
```
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs]: Ideally it should be using IFileSystem and IProcessServices

**Comment by MikhailArkhipov** [comment]
👍

File: src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs (hunk 2 above)
Comment (int19h) [inline src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs]: This entire class needs to be refactored to do that (probably by making it non-static, and taking `IFileSystem` etc in constructor). I'll file an issue.

**Pull request merged**